### ISSUE TITLE
safeExecuteWithValidation raising SAMP-Exception dialog

### ISSUE THREAD

**Issue by ghost** [SAMPProjects/Open-SAMP-API#35]
E.g.
`WRITE(int(Utils::SafeBlock::safeExecuteWithValidation([&](){
		Rendering::Renderer::sharedRenderer().getAs<Rendering::Box>(id)->setBoxHeight(height);
	})));`

id is invalid, a nullptr is returned and is used. Normally the try/catch should get it, but SAMP raise an exception dialog before. The game is running and not crashed, so the safeExecuteWithValidation works.

![Bug](https://cloud.githubusercontent.com/assets/2185647/24071839/b55fcde8-0bdb-11e7-8805-6db1de9fa0cb.png)

**Comment by agrippa1994** [comment]
Is this bug reproducible? If the exception is caught by the exception handler, then it must be somewhere else. Maybe the WRITE macro itself fails ...

**Comment by ghost** [comment]
https://github.com/shadowlif/Open_SAMP_API_ReproduceableCrash (also as Release)
Maybe it's crashing because of an missing mutex here https://github.com/SAMPProjects/Open-SAMP-API/blob/master/src/Open-SAMP-API/Game/Rendering/Renderer.cpp#L38

**Comment by agrippa1994** [comment]
That might be a reason because C++ STL containers are not thread safe. Does it still crash if you add the missing mutex there?

**Comment by ghost** [comment]
Nope, also I thought about it. Is `safeExecuteWithValidation` even catching an exception?
Maybe not because SAMP set the UnhandledExceptionHandler.
